src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/Option.cs

```diff
@@ -0,0 +1,302 @@
+using System;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common
+{
+    /// <summary>
+    /// Represents an optional result.
+    /// </summary>
+    /// <typeparam name="T">The result type.</typeparam>
+    public interface IOption<T> : IEquatable<IOption<T>>
```

Comment:
if you have `IEquatable` you should also be overriding the `Equals(object)`/`GetHashCode` methods

Reply:
Done.